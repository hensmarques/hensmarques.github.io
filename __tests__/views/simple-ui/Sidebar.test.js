import React from 'react';
import { Sidebar } from '../../../src/views/simple-ui/Sidebar';

describe('<Sidebar/>', () => {
  it('renders and matches the snapshot', () => {
    const logout = jest.fn();
    const wrapper = shallow(<Sidebar logout={logout} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('fires toggleSidebar callback when close icon is clicked', () => {
    const logout = jest.fn();
    const toggleSidebar = jest.fn();
    const wrapper = shallow(<Sidebar logout={logout} toggleSidebar={toggleSidebar} />);
    expect(toggleSidebar).not.toHaveBeenCalled();
    wrapper.find('.sidebar-icon-close').simulate('click');
    expect(toggleSidebar).toHaveBeenCalledTimes(1);
  });

  it('fires logout callback when Logout link is clicked', () => {
    const logout = jest.fn();
    const wrapper = shallow(<Sidebar logout={logout} />);
    expect(logout).not.toHaveBeenCalled();
    wrapper.find('a').simulate('click');
    expect(logout).toHaveBeenCalledTimes(1);
  });
});
