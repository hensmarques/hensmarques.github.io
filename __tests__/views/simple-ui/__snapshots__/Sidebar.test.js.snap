// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar/> renders and matches the snapshot 1`] = `
<ul
  className="sidebar"
>
  <i
    className="far fa-times fa-2x sidebar-icon-close"
    onClick={[Function]}
  />
  <li
    className="logo-container"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/home"
    >
      <img
        src="test-file-stub"
      />
    </NavLink>
  </li>
  <li
    className="item nav-link"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/home"
    >
      <i
        className="fas fa-home"
      />
      Home
    </NavLink>
  </li>
  <li
    className="item nav-link"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/buy-sell"
    >
      <i
        className="fas fa-exchange"
      />
      Buy / Sell
    </NavLink>
  </li>
  <li
    className="item nav-link"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/my-accounts"
    >
      <i
        className="fas fa-university"
      />
      My Accounts
    </NavLink>
  </li>
  <li
    className="item nav-link"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/trade"
    >
      <i
        className="fas fa-chart-line"
      />
      Pro Trade
    </NavLink>
  </li>
  <li
    className="item balances-container"
  >
    <Connect(Balances) />
  </li>
  <li
    className="item nav-link"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/settings"
    >
      <i
        className="fas fa-cog"
      />
      Settings
    </NavLink>
  </li>
  <li
    className="item nav-link"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/help"
    >
      <i
        className="fas fa-question-circle"
      />
      Help
    </NavLink>
  </li>
  <li
    className="item nav-link"
  >
    <a
      onClick={[Function]}
    >
      <i
        className="fas fa-sign-out-alt"
      />
       
      Logout
    </a>
  </li>
</ul>
`;
